iceoryx_posh/source/roudi/process_manager.cpp

```diff
@@ -533,6 +533,101 @@ void ProcessManager::addPublisherForProcess(const RuntimeName_t& name,
         [&]() { LogWarn() << "Unknown application " << name << " requested a PublisherPort."; });
 }
 
+
+void ProcessManager::addClientForProcess(const RuntimeName_t& name,
+                                         const capro::ServiceDescription& service,
+                                         const popo::ClientOptions& clientOptions,
+                                         const PortConfigInfo& portConfigInfo) noexcept
+{
+    searchForProcessAndThen(
+        name,
+        [&](Process& process) { // create a ClientPort
+            auto segmentInfo = m_segmentManager->getSegmentInformationWithWriteAccessForUser(process.getUser());
+
+            if (!segmentInfo.m_memoryManager.has_value())
+            {
+                // Tell the app no writable shared memory segment was found
+                runtime::IpcMessage sendBuffer;
+                sendBuffer << runtime::IpcMessageTypeToString(runtime::IpcMessageType::ERROR);
+                sendBuffer << runtime::IpcMessageErrorTypeToString(
+                    runtime::IpcMessageErrorType::REQUEST_CLIENT_NO_WRITABLE_SHM_SEGMENT);
+                process.sendViaIpcChannel(sendBuffer);
+                return;
+            }
+
+            m_portManager
+                .acquireClientPortData(
+                    service, clientOptions, name, &segmentInfo.m_memoryManager.value().get(), portConfigInfo)
+                .and_then([&](auto& clientPort) {
+                    // send ClientPort to app as a serialized relative pointer
+                    auto offset = rp::BaseRelativePointer::getOffset(m_mgmtSegmentId, clientPort);
```

Comment:
you could name it `relativePtrToClientPort` and then you do not need the comment.